class/defaults.yml

```diff
@@ -38,6 +38,19 @@ parameters:
       replicaLabels:
         - prometheus_replica
         - rule_replica
+    queryRbacProxy:
+      serviceName: ${_instance}-auth
+      enabled: false
+      port: 8080
+      redirectUri: https://${thanos:ingress:host}
+    ingress:
+      enabled: false
+      serviceName: ${_instance}-auth
+      servicePort: ${thanos:queryRbacProxy:port}
+      annotations:
+        cert-manager.io/cluster-issuer: letsencrypt-production
+      secretName: ${_instance}-tls
```

Comment:
Hm, fully exposing the Ingress object also means exposing arrays in TLS and backend configs. That would make it harder to configure hostnames via hierarchy.

I agree it's currently half-baked. Atm I'm opting for minimal config with only hostname, labels and annotations and make the `ingress` param a subkey of `queryRbacProxy` itself. WDYT?

Reply:
I'm happy to have a minimal config in this particular case. Adding the parameter to `queryRbacProxy` is a good option to avoid the accessible-to-the world querier setup which is otherwise easily possible.